File: src/main/java/com/murex/tbw/domain/Country.java
package com.murex.tbw.domain;

public final class Country {
    private final String name;
    private final Currency currency;
    private final Language officialLanguage;

    public Country(String name, Currency officialCurrency, Language officialLanguage) {
        this.name = name;
        this.currency = officialCurrency;
        this.officialLanguage = officialLanguage;
    }
}

File: src/cpp/lib/Country.cpp
/*******************************************************************************
 *
 * Copyright (c) {2003-2019} Murex S.A.S. and its affiliates.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the MIT License
 * which accompanies this distribution, and is available at
 * https://opensource.org/licenses/MIT
 *
 *******************************************************************************/

#include "domain/country/Country.h"

domain::country::Country::Country(const std::string& name, Currency currency, Language language): name_(name)
                                                                                                  , currency_(currency)
                                                                                                  , language_(language)
{
}
